New Tier 3 caste: Behemoth

lKiseki opened the pull request:
### BALANCE IS HARD AND THIS IS NOT FINISHED, IF YOU THINK THIS IS UNBALANCED LEAVE CONSTRUCTIVE FEEDBACK AND I WILL READ IT.
## About The Pull Request
Design document attached below, has complete information on abilities and other design choices.
https://docs.google.com/document/d/1QRJrliMgpVGgMffDMMOwNzTPhrHMYJGbWgdwkIjcLRs/edit?usp=sharing

A quick rundown or TL;DR, nonetheless;
- **Roll:** Curl up into a ball, gaining drastically increased movement speed. Cannot slash nor use abilities while this is active. Has a wind-up and a cooldown, making it a bad choice in active combat. Use it as a opener or as a way to traverse maps.
- **Landslide:** Charge forward, affecting a 3x1 area in front of you. Causes damage to _anything_ (yes, anything) caught in your path, including walls, objects, etc. Hitting a rock spawned by Earth Riser throws it as a projectile. Primal Wrath changes its functionality, allowing you to charge three times in quick succession.
- **Earth Riser:** Create a solid structure that can serve as cover. Clickdrag the rock to a target tile to launch it as a projectile. Interacts with Landslide, as specified above. Seismic Fracture will cause this structure to reflect the same attack, albeit with a delay and reduced range. Primal Wrath allows you to create three at a time. Use right click on the ability or the alternate use keybind to destroy any existing structures, from oldest to newest.
- **Seismic Fracture:** 3x3 diamond-shaped AoE, deals damage to those caught in its range. Interacts with Earth Riser by reflecting this same ability at its location. Primal Wrath greatly increases its range, but its effects are not applied if reflected by Earth Riser.
- **Primal Wrath:** Primordial ability. Uses a resource called Wrath, which you gain by being attacked and by consistently participating in combat. Increases default damage and movement speed, and upgrades most of your abilities. If you would be knocked into crit while this ability is active, Wrath will instead be used to deny incoming damage.

## Why It's Good For The Game
New content good.

## Changelog
:cl: Lewdcifer
add: Adds the Behemoth, a new tier 3 caste.
/:cl:

novaepee commented:
it's here
finally here

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/castedatum_behemoth.dm:
```diff
@@ -0,0 +1,168 @@
+	// *** Defense *** //
+	sunder_recover = 1.0
+	soft_armor = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 130, BIO = 130, FIRE = 50, ACID = 50)
+	hard_armor = list(MELEE = 25, BULLET = 25, LASER = 25, ENERGY = 25, BOMB = 0, BIO = 0, FIRE = 0, ACID = 0)
```
Todo talk about this

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/update_icons.dm:
```diff
@@ -150,6 +151,11 @@
 	icon = 'icons/Xeno/wound_overlays.dmi'
 	vis_flags = VIS_INHERIT_DIR
 
+/atom/movable/vis_obj/xeno_wounds/Initialize(mapload, datum/source)
+	. = ..()
+	if(isxenobehemoth(source))
+		icon = 'icons/Xeno/wound_overlays_3x3.dmi'
+
```
Why not just make a castedatum var like we have for the wound icon state but use it for icon instead?

lKiseki replied:
Sure. Donezo.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:
```diff
@@ -215,6 +215,12 @@
+	// *** Behemoth Abilities ***
+	/// The maximum amount of Wrath that we can have stored at a time.
+	var/wrath_max = 0
+	/// The icon state of the Wrath bar.
+	var/wrath_icon_state = "wrath"
```
This is not changed anywhere no idea why its a var

lKiseki replied:
This is because the Wrath bar is separate from health and plasma, not unlike a third bar. This is necessary so as to define the icon for that bar, because it needs specific DMI offsets to locate it besides the other bars.

lKiseki replied:
I see what you mean. Donezo.

TiviPlus reviewed (changes requested):
First pass

lKiseki commented on code/datums/keybinding/xeno.dm:
```diff
@@ -914,3 +914,39 @@
+	name = "Roll"
+	full_name = "Behemoth: Roll"
+	description = "Curl up into a ball, sacrificing some offensive capabilities in exchange for greater movement speed."
+	keybind_signal =COMSIG_XENOABILITY_LANDSLIDE
```
Fixed.

lKiseki commented on code/game/data_huds.dm:
```diff
@@ -465,6 +466,19 @@
 		holder.icon_state = "[xeno_caste.plasma_icon_state][amount]"
 
 
+/mob/living/carbon/xenomorph/proc/hud_set_wrath()
+	if(!xeno_caste) // usually happens because hud ticks before New() finishes.
+		return
```
No reason not to. Donezo.

lKiseki commented on code/game/turfs/closed.dm:
```diff
@@ -22,6 +22,16 @@
 	icon_state = "rock"
 	open_turf_type = /turf/open/floor/plating/ground/desertdam/cave/inner_cave_floor
 	minimap_color = NONE
+	/// List of possible messages a Behemoth gets when they click this object in help intent.
+	var/possible_eating_messages = list("We eat away at the stone. It tastes good, as expected of our primary diet.",
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/attack_alien.dm:
```diff
@@ -35,6 +35,8 @@
 	return TRUE
 
 /mob/living/proc/can_xeno_slash(mob/living/carbon/xenomorph/X)
+	if(status_flags & INCORPOREAL)
+		return FALSE
 	return TRUE
```
Fixed.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+#define isearthpillar(A) (istype(A, /obj/structure/earth_pillar))
+
```
Moved it to the istype define file.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	icon = 'icons/xeno/Effects.dmi'
+	icon_state = "generic_warning"
+	layer = BELOW_MOB_LAYER
+	color = "#FFD800"
```
Sure. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	animate(alpha = 0, time = 0.5 SECONDS, easing = CIRCULAR_EASING|EASE_OUT)
+
+/obj/effect/temp_visual/behemoth/warning/enhanced
+	color = "#FF0000"
```
Sure. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	if(!landslide_action?.ability_active)
+		owner.dir = get_dir(owner, target)
+
+/// Stomps anyone on the same tile as the Behemoth, damaging them and knocking them down.
+/datum/action/xeno_action/activable/behemoth/proc/do_stomp(turf/target_turf)
+	if(!target_turf)
+		return
+	playsound(target_turf, 'sound/effects/behemoth/behemoth_stomp.ogg', 30, TRUE)
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	for(var/mob/living/affected_living in target_turf)
+		if(xeno_owner.issamexenohive(affected_living) || affected_living.stat == DEAD)
+			continue
+		affected_living.emote("scream")
+		shake_camera(affected_living, max(1, BEHEMOTH_STOMP_KNOCKDOWN_DURATION), 1)
+		affected_living.Knockdown(BEHEMOTH_STOMP_KNOCKDOWN_DURATION)
+		affected_living.apply_damage(xeno_owner.xeno_caste.melee_damage / 2, BRUTE, blocked = MELEE)
+
+/// Warns nearby players, in any way or form, of the incoming ability and the range it will affect.
+/// Accepts lists, applying its effects to the contents.
+/datum/action/xeno_action/activable/behemoth/proc/do_warning(list/turf/target_turfs, duration, enhanced, action)
```
Donezo, as suggested by Tivi here and in Discord.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+
+// This ability is exclusively here to house some stuff as a parent ability, and therefore reduce the amount of repeat code.
+/datum/action/xeno_action/activable/behemoth
+	/// The duration of the ability's wind-up process.
+	var/wind_up_duration = 1 SECONDS
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	var/ability_active = FALSE
+	/// The sound that is played in ability warnings.
+	var/warning_sound = 'sound/effects/behemoth/behemoth_rumble.ogg'
+	/// References the owner's Landslide action, if any.
+	var/datum/action/xeno_action/activable/behemoth/landslide/landslide_action
+	/// References the owner's Earth Riser ability, if any.
+	var/datum/action/xeno_action/activable/behemoth/earth_riser/earth_riser_action
+	/// References the owner's Landslide action, if any.
+	var/datum/action/xeno_action/activable/behemoth/seismic_fracture/seismic_fracture_action
+	/// References the owner's Primal Wrath action, if any.
+	var/datum/action/xeno_action/primal_wrath/primal_wrath_action
```
As suggested by Tivi here and in Discord, the parent ability has been removed, along with these.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	var/wind_up_duration = 1 SECONDS
+	/// Whether this ability is currently active or not.
+	var/ability_active = FALSE
+	/// The sound that is played in ability warnings.
+	var/warning_sound = 'sound/effects/behemoth/behemoth_rumble.ogg'
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+/datum/action/xeno_action/activable/behemoth
+	/// The duration of the ability's wind-up process.
+	var/wind_up_duration = 1 SECONDS
+	/// Whether this ability is currently active or not.
+	var/ability_active = FALSE
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+			continue
+		new warning_type(target_turf, duration)
+
+/datum/action/xeno_action/activable/behemoth/succeed_activate(plasma_cost_override)
+	if(QDELETED(owner))
+		return
+	var/ability_cost = plasma_cost
+	if(plasma_cost_override)
+		ability_cost = plasma_cost_override
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	if(primal_wrath_action?.ability_active)
+		xeno_owner.wrath_stored = max(0, xeno_owner.wrath_stored - ability_cost)
+		return
+	if(ability_cost > 0)
+		xeno_owner.use_plasma(plasma_cost)
```
Donezo, as suggested by Tivi here and in Discord.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		clear_cooldown()
+	add_cooldown()
+	if(length(enhanced_turfs))
+		enhanced_turfs -= enhanced_turfs
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		turfs_list += turf_to_check
+	return turfs_list
+
+/// Moves the user in the specified direction. This simulates movement by using step() and repeatedly calling itself.
+/// Will repeatedly check a 3x1 rectangle in front of the user, applying its effects to valid targets and stopping early if the path is blocked.
+/// - owner_turf: The turf where the owner is.
+/// - direction: The direction to move in.
+/// - damage: The damage we will deal to valid targets.
+/// - which_step: Used to determine the initial positioning of visual effects.
```
Roger that. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		if(3) // During the target selection phase for the Primal Wrath version of this ability, no selection was made.
+			owner.balloon_alert(owner, "No selection made, cancelled")
+
+/// Applies several effects to a living target.
+/// - living_target: The targeted living mob.
+/// - direct_turf: If specified, living targets in this turf will receive additional effects.
+/// - damage: The damage inflicted by related effects.
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	living_target.do_jitter_animation(jitter_loops = 2)
+	living_target.apply_damage(damage * LANDSLIDE_CHARGE_DAMAGE_ADJACENT_MODIFIER, BRUTE, blocked = MELEE)
+
+/// Applies several effects to an object. Effects differ based on the object's type, with unique interactions for mechas and Earth Pillars.
+/// - object_target: The targeted object.
+/// - damage: The damage inflicted by related effects.
+/// - ending: If TRUE, applies different effects to the target.
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		playsound(object_turf, 'sound/effects/meteorimpact.ogg', 30, TRUE)
+		object_target.deconstruct(FALSE)
+
+/// Applies several effects to a target turf, and its contents if applicable. Effects differ based on the turf type and its contents, with unique interactions for walls
+/// and objects.
+/// - turf_target: The targeted turf.
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+				continue
+			hit_object(affected_object, damage, TRUE)
+
+/// Runs any checks associated to the Enhanced version of this ability, ending it early if conditions are met.
+/// - times_called: Counter for the amount of times this proc has been called. This is used to determine the amount of time spent during the selection phase,
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		return
+	addtimer(CALLBACK(src, PROC_REF(enhanced_check_charge), times_called), LANDSLIDE_ENHANCED_WAIT_DELAY)
+
+/// Prepares selections made during the selection phase of this ability. Selected turfs are processed and adapted to the ability for their later usage.
+/// If the maximum amount of selections has been reached, this proc will end the selection phase and execute the charge.
+/// - source: References the source of this proc. Usually the owner.
+/// - selected_atom: The atom that was selected by the user. This is later converted into a turf.
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	animate(pixel_y = initial(xeno_owner.pixel_y), time = animation_time / 2, flags = CIRCULAR_EASING|EASE_IN)
+	playsound(xeno_owner, 'sound/effects/behemoth/landslide_enhanced_charge.ogg', 15, TRUE)
+
+/// Moves the user between the selections previously made. This simulates movement by using step() and repeatedly calling itself.
+/// Will repeatedly check a 3x1 rectangle in front of the user, applying its effects to valid targets and stopping early if the path is blocked.
+/// - direction: The direction to move in.
+/// - damage: The damage we will deal to valid targets.
+/// - speed: The speed at which we move. This is reduced when we're nearing our destination, to simulate a slow-down effect.
+/// - steps_to_take: The amount of steps needed to reach our destination. This is used to determine when to move on to the next selection.
+/// - which_charge: Which charge, or which selection, are we currently executing.
+/// - steps_taken: The amount of steps we have taken. This is used to determine when to move on to the next selection.
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		new /obj/effect/temp_visual/behemoth/landslide/dust/charge(owner_turf, direction)
+	addtimer(CALLBACK(src, PROC_REF(enhanced_do_charge), direction, damage, speed, steps_to_take, which_charge), speed)
+
+/// Handles everything necessary before beginning the next charge.
+/// - direction: The direction we will move in.
+/// - damage: The damage we will deal in related effects.
+/// - which_charge: Which charge, or which selection, we will execute.
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+					easing = CIRCULAR_EASING|EASE_OUT, flags = ANIMATION_END_NOW)
+				animate(pixel_y = initial(affected_living.pixel_y), time = SEISMIC_FRACTURE_PARALYZE_DURATION / 2, easing = CIRCULAR_EASING|EASE_IN)
+				addtimer(CALLBACK(src, PROC_REF(living_landing), affected_living), SEISMIC_FRACTURE_PARALYZE_DURATION)
+			if(isobj(affected_atom))
```
Sure. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+/datum/action/xeno_action/activable/behemoth/seismic_fracture
+	name = "Seismic Fracture"
+	ability_name = "Seismic Fracture"
+	action_icon_state = "4"
```
Yeah, these are temporary. Kuro's gonna do the ability icons and right now I'm using placeholders. Gonna leave this unresolved until I change this.

lKiseki replied:
Implemented new ability icons.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		target_turf, extra_turfs, turfs_to_attack, enhanced, SEISMIC_FRACTURE_ENHANCED_ATTACK_RADIUS, SEISMIC_FRACTURE_ENHANCED_ATTACK_RADIUS - SEISMIC_FRACTURE_ATTACK_RADIUS), \
+		wind_up + SEISMIC_FRACTURE_ENHANCED_DELAY)
+
+/// Handles the additional attacks caused by Primal Wrath. These are done iteratively rather than instantly, with a delay inbetween.
+/// - origin_turf: The starting turf.
+/// - extra_turfs: Any additional turfs that should be handled.
+/// - excepted_turfs: Turfs that should be excepted from this proc.
+/// - enhanced: Whether this is enhanced by Primal Wrath or not.
+/// - range: The range to cover.
```
Fixed.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PRIMAL_WRATH,
+	)
+	/// The xeno owner of this ability.
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
```
I am a user of black magicks, Tivi. I cannot be stopped...
Jokes aside, it actually (somehow) compiled, but I removed it since it's not necessary.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	)
+	/// The xeno owner of this ability.
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
+	/// References the owner's Earth Riser ability, if any.
+	var/datum/action/xeno_action/activable/behemoth/earth_riser/earth_riser_action
```
Donezo as suggested.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		addtimer(CALLBACK(world_plane, TYPE_PROC_REF(/atom, remove_filter), "primal_wrath"), resolve_time)
+		return
+	addtimer(CALLBACK(src, PROC_REF(ability_check), affected_living, xeno_source), 0.1 SECONDS)
+
+/// Resets decay, and gives the owner an amount of Wrath proportional to the damage taken.
+/// - source: The source of this proc.
+/// - amount: The RAW amount of damage taken.
+/// - amount_mod: If provided, this list includes modifiers applied to the damage. This, for example, can be useful for reducing the damage.
```
Fixed.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	. = ..()
+	animate(src, alpha = 0, time = duration, easing = CIRCULAR_EASING|EASE_OUT)
+
+// This is a structure that can be created by the Earth Riser ability. See it for specifics.
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	'sound/effects/behemoth/earth_pillar_hit_6.ogg')
+	destroy_sound = 'sound/effects/behemoth/earth_pillar_destroyed.ogg'
+	/// List of possible messages a Behemoth gets when they click this object in help intent.
+	var/possible_eating_messages = list("We eat away at the stone. It tastes good, as expected of our primary diet.",
+	"Mmmmm... Delicious rock. A fitting meal for the hardiest of creatures.",
+	"This boulder -- its flavor fills us with glee. Our palate is thoroughly satisfied.",
+	"These minerals are tasty! We want more!",
+	"Eating this stone makes us think; is our hide tougher? It is. It must be.",
+	"A delectable flavor. Just one bite is not enough...",
+	"One bite, two bites... why not just finish the whole rock?",
+	"The stone. The rock. The boulder. The crag. Its name matters not when we consume it.",
+	"We're eating this boulder. It has other uses... Is this really a good idea?",
+	"Delicious, delectable, simply exquisite. Just a few more minerals and it'd be perfect...")
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	'sound/effects/behemoth/earth_pillar_hit_4.ogg',
+	'sound/effects/behemoth/earth_pillar_hit_5.ogg',
+	'sound/effects/behemoth/earth_pillar_hit_6.ogg')
+	destroy_sound = 'sound/effects/behemoth/earth_pillar_destroyed.ogg'
+	/// List of possible messages a Behemoth gets when they click this object in help intent.
```
Oop, my bad. Fixed.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+		return
+	warning_flashes--
+	addtimer(CALLBACK(src, PROC_REF(seismic_fracture)), 1 SECONDS)
+	animate(src, color = "#FF6A00", time = 0.5 SECONDS, easing = CIRCULAR_EASING|EASE_OUT, flags = ANIMATION_PARALLEL)
+	animate(color = "#FFFFFF", time = 0.5 SECONDS, easing = CIRCULAR_EASING|EASE_IN, flags = ANIMATION_PARALLEL)
```
Sure. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	icon_state = "earth_pillar"
+	sound_hit = 'sound/effects/behemoth/earth_pillar_destroyed.ogg'
+	sound_bounce = 'sound/effects/behemoth/earth_pillar_destroyed.ogg'
+	ping = null
```
Dunno which one of these three lines you mean, but I'll provide an explanation for all three:
`sound_hit`: This is the sound played when the projectile hits someone. I did this intentionally.
`sound_bounce`: Unneeded since this is not a bouncing projectile. Removed.
`ping`: This is an animation that plays when the projectile hits something. It is nulled because it's otherwise an acid-looking icon, which is unfitting for this.

lKiseki replied:
Talked with Tivi in Discord about this, there was a misunderstanding regarding what `ping` did. Resolved.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+/datum/action/xeno_action/ready_charge/behemoth_roll
+	name = "Roll"
+	ability_name = "Roll"
+	action_icon_state = "4"
```
Yeah, these are temporary. Kuro's gonna do the ability icons and right now I'm using placeholders. Gonna leave this unresolved until I change this.

lKiseki replied:
Kuro gave me the icons and I've implemented them. Resolving.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	duration = 2 SECONDS
+	layer = ABOVE_ALL_MOB_LAYER
+
+/obj/effect/temp_visual/behemoth/earth_pillar/Initialize(mapload)
+	. = ..()
+	layer += 0.01
```
This is probably unnecessary in the first place. Removed.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	QDEL_NULL_IN(src, particle_holder, 1.6 SECONDS)
+	do_jitter_animation(jitter_loops = 5)
+
+/obj/structure/earth_pillar/Destroy()
+	playsound(src, destroy_sound, 40, TRUE)
+	if(earth_riser_action && (src in earth_riser_action.active_pillars))
+		earth_riser_action.active_pillars -= src
+	return ..()
```
Ehh, I can put it there myself. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+// - Help intent as a Behemoth will trigger an easter egg. Does nothing, just fluff.
+/obj/structure/earth_pillar/attack_alien(mob/living/carbon/xenomorph/xeno_user, isrightclick = FALSE)
+	var/current_turf = get_turf(src)
+	switch(xeno_user.a_intent)
+		if(INTENT_HARM)
+			if(attacks_to_destroy <= 1)
```
Sure. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1463 @@
+	if(length(enhanced_warnings))
+		enhanced_warnings -= enhanced_warnings
+	switch(reason)
+		if(1) // The user manually cancelled the ability at some point during its use.
+			owner.balloon_alert(owner, "Cancelled")
+		if(2) // During the charge, the user did not have enough plasma to satisfy the charge's cost.
+			owner.balloon_alert(owner, "Insufficient plasma")
+		if(3) // During the target selection phase for the Primal Wrath version of this ability, no selection was made.
```
Sure. Donezo.

FosterGrime-Heymynameslouie commented:
I eat rocks for breakfast
I eat rocks for lunch
I eat rocks for dinner

Al-1ce commented:
Interesting.

mc-oofert commented:
i think this would be better as T4 this shit has a major amount of armor

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+	target_turfs += get_step(xeno_owner, turn(direction, 45))
+	target_turfs += get_step(xeno_owner, turn(direction, -45))
+	for(var/turf/target_turf AS in target_turfs)
+		for(var/atom/movable/affected_atom in target_turf)
```
```suggestion
		for(var/atom/movable/affected_atom AS in target_turf)
```

lKiseki replied:
Donezo.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+		var/mob/living/carbon/xenomorph/xeno_owner = owner
+		xeno_owner.balloon_alert(xeno_owner, "No active pillars")
+		return
+	var/obj/structure/earth_pillar/oldest_pillar = active_pillars[1]
+	new /obj/effect/temp_visual/behemoth/earth_pillar/broken(get_turf(oldest_pillar))
+	QDEL_NULL(oldest_pillar)
```
```suggestion
	var/obj/structure/earth_pillar/oldest_pillar = popleft(active_pillars)
	new /obj/effect/temp_visual/behemoth/earth_pillar/broken(get_turf(oldest_pillar))
	qdel(oldest_pillar)
```

lKiseki replied:
Donezo.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+	while(length(active_pillars) > maximum_pillars)
+		var/obj/structure/earth_pillar/oldest_pillar = active_pillars[1]
+		new /obj/effect/temp_visual/behemoth/earth_pillar/broken(get_turf(oldest_pillar))
+		QDEL_NULL(oldest_pillar)
```
```suggestion
		qdel(oldest_pillar)
```

lKiseki replied:
Donezo.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+	do_jitter_animation(jitter_loops = 5)
+
+/obj/structure/earth_pillar/Destroy()
+	var/mob/living/carbon/xenomorph/xeno_owner = owner
```
Why are you casting owner here instead of in the defenition on the object lol

lKiseki replied:
Fixed.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+	soft_armor = list(MELEE = 50, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 0, BIO = 100, FIRE = 100, ACID = 0)
+	destroy_sound = 'sound/effects/behemoth/earth_pillar_destroyed.ogg'
+	coverage = 128
+	/// The owner of this object.
+	var/owner
```
If the behemoth gets deleted all these pillars will not be deleted so that owner = null and will cause a hard delete

You can fix this by QDEL_LIST the pillars list + add owner = null to pillar Destroy when you lose the ability I guess?

lKiseki replied:
Sure, donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/castedatum_behemoth.dm:
```diff
@@ -0,0 +1,180 @@
+/datum/xeno_caste/behemoth/primordial
+	upgrade_name = "Primordial"
+	upgrade = XENO_UPGRADE_FOUR
+	primordial_message = "We are nature's wrath made manifest."
```
no

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+
+/// Warns nearby players, in any way or form, of the incoming ability and the range it will affect.
+/proc/do_warning(owner, list/turf/target_turfs, duration, enhanced, action)
+	if(!owner || !length(target_turfs) || !duration)
+		return
```
Sure. Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+/proc/do_warning(owner, list/turf/target_turfs, duration, enhanced, action)
+	if(!owner || !length(target_turfs) || !duration)
+		return
+	var/warning_type = enhanced? /obj/effect/temp_visual/behemoth/warning/enhanced : /obj/effect/temp_visual/behemoth/warning
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+		shake_camera(xeno_owner, 1, 0.5)
+		addtimer(CALLBACK(src, PROC_REF(end_charge)), LANDSLIDE_ENDING_COLLISION_DELAY)
+		return
+	which_step =! which_step
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/behemoth.dm:
```diff
@@ -0,0 +1,66 @@
+	user_unbuckle_mob(source, source)
+
+/mob/living/carbon/xenomorph/behemoth/primordial/Stat()
+	. = ..()
```
Fixed.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1553 @@
+	/// Whether this ability is currently active or not.
+	var/ability_active = FALSE
+	/// List containing possible sound effects that are played during this ability.
+	var/list/possible_step_sounds = list(
+		'sound/effects/alien_footstep_large1.ogg',
+		'sound/effects/alien_footstep_large2.ogg',
+		'sound/effects/alien_footstep_large3.ogg')
+	/// List containing the selected turfs for the Enhanced version of this ability.
```
Sure. Will solve in the following commits.

Kuro020 reviewed (approved):
behemoth my beloved

zee11zee commented:
The charge broke weeds, that seems like a big problem.

seif1 commented:
> 

It also breaks sticky resin, and I assume any other xeno object

while charging, you can be resting

Kuro020 commented:
Some things I saw.
-Primo is bugged, you cant get wrath
-Being able to cancel the charge is pretty necessary
-It really needs the roll mode, having half of the kit makes it pretty guh

imsxz commented:
![image](https://github.com/tgstation/TerraGov-Marine-Corps/assets/27370365/dd8ba7d4-3ba7-4b92-8f56-ed2c63e55441)
you know somethings up when the dev himself switches castes mid round....

abilities are too slow to be able to hit confirm your combos, too long of cooldowns to make you want to use all of your buttons at once. lots of downtime between abilities in general feels bad, no filler and not fast enough to stick around slashing.

having mechanics to reduce ability cooldowns would be nice, maybe refreshing charge cooldown if its used to kick a rock or something.

will probably feel a lot better to play with roll added and more people playing other castes instead of 10 behemoths. i imagine gas castes and such will make setting up combos a lot easier

DeltaFire15 commented:
Pretty fun even when half-broken, which only points to what it'll be when done.
One bug I noticed was rocks sometimes 'multi-detonating' (exploding twice) if hit by your explosion; They'll effectively explode, delete, then explode again. This is likely from their first explosion triggering before they qdel and queueing the boom again, or something the like I'd assume. This happens really inconsistently though.

chizzy376 commented:
> Pretty fun even when half-broken, which only points to what it'll be when done. One bug I noticed was rocks sometimes 'multi-detonating' (exploding twice) if hit by your explosion; They'll effectively explode, delete, then explode again. This is likely from their first explosion triggering before they qdel and queueing the boom again, or something the like I'd assume. This happens really inconsistently though.

if you do this right you can make an infinite detonation by recasting the same rock

iforgotmeword commented:
ah yes a other xeno to assist in gangraping tad ;-; 
whatever i will find a counter like i always do

lKiseki commented:
Thanks for the feedback, everyone. Right now I'll be taking a break since I've been nonstop coding Behemoth for more than a week by now, but after that I'll do what I can to tackle this feedback.

Pariah919 commented:
se acabo

Kuro020 commented:
> se acabo

not yet

lKiseki commented:
The return.
We were never gone, ladies and gentlemen.

TiviPlus commented:
COMSIG_PARENT_QDELETING got renamed to COMSIG_QDELETING
and flags_pass also got renamed

Aporhtonoma commented:
pretty iffy about giving it hard armor and wrath being a rav endure(depends on how many max points you do give it though)
also does the stone pillar thing stay and is infinetly placeable or does it have a cap 
but otherwise it legitimately looks both fun to play and fight against

lKiseki commented:
> pretty iffy about giving it hard armor and wrath being a rav endure(depends on how many max points you do give it though) also does the stone pillar thing stay and is infinetly placeable or does it have a cap but otherwise it legitimately looks both fun to play and fight against

The stone pillar has a normal limit of one per Behemoth, increased to 3 while Wrath is active. It otherwise has an indefinite duration, but you have a few options to get rid of it if needed.

DeltaFire15 commented:
> pretty iffy about giving it hard armor

Hard armor got reworked to be basically "anti-AP" instead of flat DR.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/xenomorph.dm:
```diff
@@ -4,7 +4,7 @@
 //Just about ALL the procs are tied to the parent, not to the children
 //This is so they can be easily transferred between them without copypasta
 
-/mob/living/carbon/xenomorph/Initialize(mapload)
+/mob/living/carbon/xenomorph/Initialize(mapload, footstep_type = FOOTSTEP_XENO_MEDIUM)
```
Donezo.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:
```diff
@@ -276,7 +283,7 @@
 	see_infrared = TRUE
 	hud_type = /datum/hud/alien
-	hud_possible = list(HEALTH_HUD_XENO, PLASMA_HUD, PHEROMONE_HUD, QUEEN_OVERWATCH_HUD, ARMOR_SUNDER_HUD, XENO_DEBUFF_HUD, XENO_FIRE_HUD)
+	hud_possible = list(HEALTH_HUD_XENO, PLASMA_HUD, WRATH_HUD, PHEROMONE_HUD, QUEEN_OVERWATCH_HUD, ARMOR_SUNDER_HUD, XENO_DEBUFF_HUD, XENO_FIRE_HUD)
```
Donezo.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/behemoth.dm:
```diff
@@ -0,0 +1,82 @@
+/mob/living/carbon/xenomorph/behemoth/resisted_against(datum/source)
+	user_unbuckle_mob(source, source)
+
+/mob/living/carbon/xenomorph/behemoth/Stat()
+	. = ..()
+	if(statpanel("Game") && xeno_caste.wrath_max > 0)
+		stat("Wrath:", "[wrath_stored] / [xeno_caste.wrath_max]")
```
Just heads up that if new statpanel is merged this would need to be tweaked

lKiseki replied:
Gotcha, I'll keep it in mind.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1462 @@
+	var/charge_damage = (xeno_owner.xeno_caste.melee_damage * xeno_owner.xeno_melee_damage_modifier) * LANDSLIDE_DAMAGE_MULTIPLIER
+	addtimer(CALLBACK(src, PROC_REF(do_charge), owner_turf, direction, charge_damage, which_step), LANDSLIDE_WIND_UP)
+
+/** Gets a list of the turfs affected by this ability, based on direction and range.
+* * origin_turf: The origin turf from which to start checking.
+* * direction: The direction to check in.
+* * range: The range in tiles to limit our checks to. */
```
```suggestion
/** Gets a list of the turfs affected by this ability, based on direction and range.
 * * origin_turf: The origin turf from which to start checking.
 * * direction: The direction to check in.
 * * range: The range in tiles to limit our checks to.
 */
```

lKiseki replied:
Fixed.

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1462 @@
+				playsound(src, 'sound/effects/behemoth/earth_pillar_destroyed.ogg', 40, TRUE)
+				new /obj/effect/temp_visual/behemoth/landslide/hit(current_turf)
+				new /obj/effect/temp_visual/behemoth/earth_pillar/broken(current_turf)
+				Destroy()
+				return TRUE
```
You use qdel(src) instead of Destroy()

TiviPlus commented on code/modules/mob/living/carbon/xenomorph/castes/behemoth/abilities_behemoth.dm:
```diff
@@ -0,0 +1,1462 @@
+/obj/structure/earth_pillar/ex_act(severity)
+	switch(severity)
+		if(EXPLODE_DEVASTATE)
+			Destroy()
```
ditto

lKiseki replied:
Fixed.

TiviPlus reviewed (approved):
rest lgtm

Pariah919 commented:
Just off a singular pass since I never bothered to read the numbers before this I think the surviablity will turn out heavily inflated

```
+ This xeno can be ridden, and allows for two riders at a time.```
Just seems like designcreep of compared to crusher allowing 1, allowing 2 riders just means people will stack a spit caste+drone to troll around and kill people.
```
+ Very tanky, boasting high health and armor, good for frontlining.```
Its pretty tanky, boasting 700 hp
```
•  Decent damage. Neither good, nor bad.```
It is inafct, average.

```
•  Behemoths can eat just about any mineral. Rocks, ores, crystals, sandbags… 
-  Loud footsteps, easily heard at a distance.```
Aesthetic is cool
```
-  Slow movement requires situational awareness to not be caught in a bad spot.
```
Weed speed still being there makes this caste surprisingly nimble at -0.2 and makes the idea that it is 'slow' laughable with it having several movement abilities to make up for its slowness outside of combat, it will not suffer from being slow due to also having incredibly high health

For an example King is slower than this, but has pheromones so it might even out but this is tankier to almost all sources of damage alongside your charge and roll and also throwing rocks

In general this is more of a base stat critique than me disliking the concept itself but I do think it has a lot of red flags in its hard numbers, personally I would either 
A. remove weed speed from it entirely so its downside of being slow is more apparent, or B. Reduce the health to 500

The weakness is supposed to be in movement but it has abilities that counteract that weakness which just leaves you with an incredibly tough to kill target with solid damage, speed and tools to wreck marines

Weedmaster77 commented:
> Weed speed still being there makes this caste surprisingly nimble at -0.2 and makes the idea that it is 'slow' laughable with it having several movement abilities to make up for its slowness outside of combat, it will not suffer from being slow due to also having incredibly high health. 

I Agree Strongly with this seems too quick when it's got so much heal and armor, and with Roll it can disengage and just runs when it's on deaths door.  

 Think the rock it gets from Earth Riser health should be based on its age if it's not already making rocks more breakable since they get used a blockade so there is a chance for marine to get out and not just be trapped.

Weedmaster77 commented:
After more play in higher pop, it seems like a better crusher in every way. With its Landslide Ability letting it sideswipe tad killing turrets, mounted weapons and anything deployed in one shot seems like a bit much, maybe its side tiles in the charge take less damage than the tile its traveling on so that engineers get a chance to even try to save or repair anything.  Also, razor wire having no effect on it like crusher make crusher pointless when you have queen and behemoth that can just charge and not get stuck or even hurt. Still think the Caste is a bit too powerful for a T3 maybe make it a T4 and make it locked to only one and let it be strong or make it just a slightly better niche crusher but not up show the crusher in its main ability.

Arturlang commented:
> Needs some more critical examination of the balance/stats of it, since the round (Round ID: 22569) i was just in saw up to 4-5 behemoths due to a mix of it's ability to stun efficiently, destroy most cades in its way, being a bit too capable in toe-to-toe battle, and ultimately having not a whole lot of counters short of "shoot it dead fast when it messes up" or catching it off-guard with electrified razorwire maybe.

I wonder given it still has plasma, and so *little* of it, that anything that removes it should be extra effective in disabling it's abilities?

lKiseki commented:
> You seem to completely stop gaining wrath after using primal wrath. Plus, it really should not be completely explosive immune.

Fixed the issue with Primal Wrath.

As for the explosion immunity, that's one thing I do not intend to change. I'll probably keep it as-is.

Strayed-Raven commented:
I like how Behemoth went from being janky in abilities to now having better utility than Crusher carry with two xenos instead of one, plus Bull mobility with roll, and the range attack potential of Warlock. As it stands there is no reason not to combo platter with a warlock and erase cades en masse.

chizzy376 commented:
Right now behemoth can very easily and safely destroy the mounted guns on top of tad with no risk by the use of his earth slam which has a massive range. This feels very unfair, as you can be scammed out of a 1000 req point investment with no counter.

ivanmixo commented:
Being able to break cades in one charge while the caste that is supposed to be dedicated to charging needs 2 is not cool.

Scarlet-Oduko commented:
just my thoughts on behemoth from being against it and around it, have  yet to actually control one so I may be a bit biased about it but here are some of my feedback points.

As mentioned above cade destroying is very funky, as well as its charge seems to completely negate razor wire. Which is funky considering even crusher has to atleast slash razor to properly charge through it.

The potential for being chained by the stuns while overstated, is still pretty high, especially with multiple around you. it feels very overwhelming both visually and mechanically. I dunno if that's the intention but the indicators being brown blends in quiet well when in the action.

I don't have too much feed back as I feel you're mostly aware of things but yeah, I think the biggest issuse it feels like its trying to fill too many roles at once. it has the tankiness of gorger while being an incredibly good CC and able to kill cades and entrenched places on par with warlock.

henriquekill9576 commented:
> 

its very much not good in cc, your sprites is 1x3 tiles wide so it makes it almost

> just my thoughts on behemoth from being against it and around it, have yet to actually control one so I may be a bit biased about it but here are some of my feedback points.
> 
> As mentioned above cade destroying is very funky, as well as its charge seems to completely negate razor wire. Which is funky considering even crusher has to atleast slash razor to properly charge through it.
> 
> The potential for being chained by the stuns while overstated, is still pretty high, especially with multiple around you. it feels very overwhelming both visually and mechanically. I dunno if that's the intention but the indicators being brown blends in quiet well when in the action.
> 
> I don't have too much feed back as I feel you're mostly aware of things but yeah, I think the biggest issuse it feels like its trying to fill too many roles at once. it has the tankiness of gorger while being an incredibly good CC and able to kill cades and entrenched places on par with warlock.

its very much horrible in cc, the sprite being 1x3 makes it impossible to click slash marines, combined with the slow speed also making it impossible to bump slash, plus the fact you'll probably spend most of your time running away after 1-2 slashes since anymore wont give you enough time to escape with your slow ass

Scarlet-Oduko commented:
> its very much horrible in cc, the sprite being 1x3 makes it impossible to click slash marines, combined with the slow speed also making it impossible to bump slash, plus the fact you'll probably spend most of your time running away after 1-2 slashes since anymore wont give you enough time to escape with your slow ass

when referring to CC, I mean Crowd control and not close combat. though that is also a good point that its sprite is massive and that affects a lot.

henriquekill9576 commented:
> > its very much horrible in cc, the sprite being 1x3 makes it impossible to click slash marines, combined with the slow speed also making it impossible to bump slash, plus the fact you'll probably spend most of your time running away after 1-2 slashes since anymore wont give you enough time to escape with your slow ass
> 
> when referring to CC, I mean Crowd control and not close combat. though that is also a good point that its sprite is massive and that affects a lot.

a i see

Weedmaster77 commented:
Seems the more and more I play against the Behemoth all I see is stun spam since all its abilities besides roll and wrath if hit with knock you over or throw you in the air and then it's just a never ending stun lock with any other xeno that is remotely close to you set up to stun you again and again making it really unfun to play against, and a team with anything more than 2 Behemoths is just stun locking the whole front. Maybe its new and it just gets spammed but that will not change the stun spam. Think the rock throw should just damage you and not stun since that is opener that is always used to start the lock down and is the easiest to land.

imsxz commented:
can someone review this so we can play it again :3

Lumipharon commented:
Being able to literally remove an entire flank of cades through a solid wall is not idea.

Lumipharon reviewed (changes requested):
Needs fixing for:

1. Damaging things through walls.
2. huge structure damage in general making it the superior seige caste
3. LOS apparently not being required for abilities
4. roll momentum is conserved while changing direction (I can only assume this is not intended?)
5. It being the tankiest caste in the game, while also being the only instant cast ranged stunner in the game

imsxz commented:
>     6. It being the tankiest caste in the game while respectably fast even without roll
> 
>     7. Being literally the most powerful stun caste in the game, with powerful, spammable ranged stuns on instant cast

i dont see whats wrong with these 2 as long as its not a siege monster. the stuns are easy to dodge if its not being combod with other xenos. ive been primarily playing marine during behemoth testmerges and ive found its pretty easy to sidestep their abilities, break line of sight with the rock to prevent the good ranged stun, and rush down the behemoth so that its forced to retreat. i think itd be silly if they were just slow turds that die if their abilities dont hit(the boiler experience)

Lumipharon commented:
The issue is not offensive. Its defensively its stuns and rock pillar make it effectively impossible to chase in a maze as long as it has abilities ready to instant cast.

imsxz commented:
cant you say that about most castes if their abilities are not on cooldown? every other t3(and t4) is definitely impossible to chase thru maze if their abilities are off cooldown. 

in my head when i think of a tank, i imagine them dying to focus fire. i dont think its unreasonable that theyre safe from getting burst down by solo marines chasing them through maze.

MemedHams commented:
> The issue is not offensive. Its defensively its stuns and rock pillar make it effectively impossible to chase in a maze as long as it has abilities ready to instant cast.

I mean, crusher is similarly lethally dangerous to pursue if it has stop/toss off cooldown, and it's also a t3

Lumipharon commented:
Ranged stun. Ranged AOE stun. and hard path blocker. Both are instant cast, but cannot be countered.
A crusher/rav etc can stun someone chasing, but they have to actually hit an ability, and actively interact with them to be able to use it successfully. I.e. there is counter play. Bemo just casts in front of it and keeps running and you can't avoid it.

There's a reason ranged stuns are very rare or restricted. Only shrike and queen have ranged instant cast stuns.

imsxz commented:
some things i noticed last night: primo is filled instantly upon taking any damage, which is about as balanced as you can imagine. felt absurdly powerful but i reckon it will be a lot less oppressive with fixed wrath generation.

also, i think theres some issues with wrath consumption . i couldnt use abilities in tanglefoot with wrath activated due to "lack of plasma" with full wrath and 0 plasma. Also, im not sure if the abilities properly consume wrath in general or of the value is very low, but i could always get in extra cheeky casts right when wrath was about to run out

primo earth risers go on cooldown as soon as you fling any of tem, so if you dont place all 3 first you'll deny yourself bonus earth risers. also, just a personal thought but it takes like 5 or 6 seconds to place all 3 of them which feels pretty clunky mid combat, dunno if it would be too strong if the mini coodlown was shorter.

would be nice if landslide could still do damage to non defensive/non deployables(i..e. tables, vending machines, window frames)

rock toss AOE nerf was deserved. even after the general nerfs, caste is still a lot of fun to play

MisterGrimm commented:
If you rage while landsliding you can cancel the landslide, allowing you to move while doing the rage roar.

henriquekill9576 commented:
wrath can easily be farmed by slashing cades/razor incase thats not intended

MisterGrimm commented:
> wrath can easily be farmed by slashing cades/razor incase thats not intended

I wouldn't say easily, since it takes about four metal cades to get 200 wrath. Although, razor is definitely easier to do it with, but I see that more as punishing a engineer for razor grenade spam.

ivanmixo commented:
> wrath can easily be farmed by slashing cades/razor incase thats not intended

Rav can do the same thing so it's fine

imsxz commented:
fracture can cancel landslide currently which feels pretty bad, since you could combo them nicely before.

one thing ive noticed is that on crash this caste feels unkillable due to the lack of players but im not sure if theres a solution to that, although i imagine crusher isnt very killable on crash either so its probably a crash issue

i kinda think the HP nerf was overkill since speed was also nerfed. i havent watched many rounds recently though so idk how much the nerfs have impacted

ivanmixo commented:
Giving it structure damage makes it the best siege caste with no contest so that's a no, we're not going through this again.

MisterGrimm commented:
> > Giving it structure damage makes it the best siege caste with no contest so that's a no, we're not going through this again.
> 
> I'm not saying to bring the same structure damage it had before, even warlock on release had insane sieging abilities and look at it now, balanced even with cade damaging aoe attack.
> 
> Make it just deal less structure damage than it used to and it should be useful while not being better than other siege castes.

Making it where only landslide could damage cades, but only at 25% original damage would probably work balance-wise. It just feels like a giant hole in logic to outright turn off structure damage for the behemoth, when all of these other castes can damage cades with their damaging abilities.

imsxz commented:
disagree with a lot of those points tbh, also id compare it to gorger before other t3 castes. its probably the single best caste for following up on queen screech and also maze trapping marines. 

i will agree tho landslide is almost laughably slow.

also i dont think structure damage is healthy for behemoths kit, especially not for fracture/earth riser

MisterGrimm commented:
Behemoths will stop on knocked over vending machines instead of passing over them.

ivanmixo commented:
Make the hive menu bigger, the t3s literally don't fit in the default width

DeltaFire15 commented:
Experience with the currently testmerged version has been pretty great.
I had some initial concerns about that lack of structure damage, however seeing how its abilities can interact with other xenos during a siege of defended locations, it is still a decent support for them (especially when primordial).
Some token structure damage on abilities would probably still be nice (it's kind weird for a charge to just kinda do nothing to a table), but it's not really a neccessity.

Mannybrado commented:
> and even gorger

he says, like gorger's entire gimmick isnt being tanky as fuck

lKiseki commented:
I'm gonna be closing this PR for the time being.

I will admit it's been hard to keep up motivation for this caste, especially considering I have to change the design again to better account the change from a sieging caste to a frontlining caste. Abilities cannot remain the same, nor am I content with half-assing this simply by doing minor changes or band-aids.

This does **NOT** mean Behemoth or that this PR are dead. They are not, and I will return to them once I have a good idea of what to do moving forward.

Until then, I'll keep this closed. A warm thank you to everyone involved with this project and the feedback I've received thus far.

lKiseki commented:
So we back in the mines.

Oscari-pog commented:
It has xegun

novaepee commented:
we so back

GrayRachnid commented:
## Feedback:

Behemoth's slash damage shouldn't be high. They have basically the same EHP(against normal rifles) as a crusher without suffering sunder consequences. They have more slash damage than a crusher and a gorger while having the capability to be a completely ranged caste. 

In addition to being a stun heavy ranged caste, it's a strong melee caste with strong mobility. 

Things I think should be changed:
- Lower slash damage to 21 or 22. For reference, crusher is 24 and gorger is 20. 
- Tweak the roll from a permanent movement to a self buff.
- Make it an armor centered tanky caste instead of a health centered caste. That's the gorger's gimmick and I don't think it should be doing it better.

lKiseki commented:
As per latest commit;
- Landslide and its enhanced version will now do minimal structure damage again.
  -  This will affect most irrelevant or weak structures, such as racks, windows and their frames (though not certain reinforced types), closets and crates, and so on. This is simply here to prevent an ability use from being wasted just because a mere rack was blocking your path.
  - Anything besides the above, such as sentries, barricades, walls, etc. **are completely unaffected by this change.**
- Fixed a bug where Landslide would endlessly loop if it was stopped by a buckled mob. The mob will now be forcibly unbuckled, then knocked down, allowing Landslide to progress as normal.
- All abilities now interact with fuel tanks, causing them to explode.
- Earth Pillars will no longer stop on the targeted tile; instead, they will travel up to ten tiles in any given direction, or until they hit something.
- Updated the Behemoth mob define to properly reflect the correct health and plasma values.
- Fixed a bug where Behemoth could vent crawl.

Looking4bros commented:
Quick, reset behemoth to the testmerge where it had near unlimited rage and was cool and fun to fight as vali!!
The distraction won't last forever!
